client/components/overviewWidget/ProductOverview.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import axios from 'axios';
+import ProductFeatures from './ProductFeatures.jsx';
+
+class ProductOverview extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      product: null,
+      slogan: '',
+      description: '',
+      features: []
+    };
+  }
+
+  componentDidMount() {}
+
+  componentDidUpdate(prevProps) {
+    if (this.props.product !== prevProps.product) {
```

Comment:
Flagging that setting your state from props is discouraged by React.  I believe this is what is causing you to have to add this logic in componentDidUpdate instead of using the props being passed down